### PR TITLE
crypto: use non-default scrypt parameters for encrypting/decrypting

### PR DESCRIPTION
**Pull request by AnnaShaleva**
This PR should fix `Tests` Github Workflow failing due to timeout while decrypting account:
```
2021-05-31T08:30:46.4452214Z goroutine 4306 [runnable]:
2021-05-31T08:30:46.4452963Z golang.org/x/crypto/scrypt.salsaXOR(0xc00034d7e8, 0xc00017d980, 0x1a0, 0x1a0, 0xc00017dcc0, 0xd0, 0xd0)
2021-05-31T08:30:46.4454249Z 	/home/runner/go/pkg/mod/golang.org/x/crypto@v0.0.0-20190701094942-4def268fd1a4/scrypt/scrypt.go:34 +0x118b
2021-05-31T08:30:46.4455237Z golang.org/x/crypto/scrypt.blockMix(0xc00034d7e8, 0xc00017d800, 0x200, 0x200, 0xc00017dc00, 0x100, 0x100, 0x8)
2021-05-31T08:30:46.4456510Z 	/home/runner/go/pkg/mod/golang.org/x/crypto@v0.0.0-20190701094942-4def268fd1a4/scrypt/scrypt.go:134 +0x258
2021-05-31T08:30:46.4457470Z golang.org/x/crypto/scrypt.smix(0xc00045fc00, 0x400, 0x400, 0x8, 0x4000, 0xc002354000, 0x400000, 0x400000, 0xc00017d800, 0x200, ...)
2021-05-31T08:30:46.4458729Z 	/home/runner/go/pkg/mod/golang.org/x/crypto@v0.0.0-20190701094942-4def268fd1a4/scrypt/scrypt.go:156 +0x4e5
2021-05-31T08:30:46.4459795Z golang.org/x/crypto/scrypt.Key(0xc00040cbe8, 0x3, 0x8, 0xc00045c012, 0x4, 0xa5, 0x4000, 0x8, 0x8, 0x40, ...)
2021-05-31T08:30:46.4461073Z 	/home/runner/go/pkg/mod/golang.org/x/crypto@v0.0.0-20190701094942-4def268fd1a4/scrypt/scrypt.go:209 +0x2bf
2021-05-31T08:30:46.4462573Z github.com/nspcc-dev/neo-go/pkg/crypto/keys.NEP2Decrypt(0xc000417000, 0x3a, 0xc000550ac8, 0x3, 0xfc478f, 0xc000426420, 0x1991820)
2021-05-31T08:30:46.4463796Z 	/home/runner/work/neo-go/neo-go/pkg/crypto/keys/nep2.go:92 +0x1b1
2021-05-31T08:30:46.4464899Z github.com/nspcc-dev/neo-go/pkg/wallet.(*Account).Decrypt(0xc0001cee70, 0xc000550ac8, 0x3, 0x0, 0x0)
2021-05-31T08:30:46.4465987Z 	/home/runner/work/neo-go/neo-go/pkg/wallet/account.go:128 +0x150
2021-05-31T08:30:46.4467405Z github.com/nspcc-dev/neo-go/pkg/consensus.NewService(0xc0009caea0, 0xc0003ae1b0, 0x1d632a0, 0xc0002af440, 0x2a, 0x64, 0x3e8, 0x4000000000000, 0xd18c2e2800, 0x20000030d40, ...)
2021-05-31T08:30:46.4468795Z 	/home/runner/work/neo-go/neo-go/pkg/consensus/consensus.go:145 +0x789
2021-05-31T08:30:46.4470240Z github.com/nspcc-dev/neo-go/pkg/network.newServerFromConstructors(0x0, 0x5, 0xa, 0xc0002922e0, 0x14, 0x0, 0x0, 0x2a00000000, 0x1, 0x0, ...)
2021-05-31T08:30:46.4471623Z 	/home/runner/work/neo-go/neo-go/pkg/network/server.go:211 +0x1002
2021-05-31T08:30:46.4472803Z github.com/nspcc-dev/neo-go/pkg/network.NewServer(0x0, 0x5, 0xa, 0xc0002922e0, 0x14, 0x0, 0x0, 0x2a00000000, 0x1, 0x0, ...)
2021-05-31T08:30:46.4473942Z 	/home/runner/work/neo-go/neo-go/pkg/network/server.go:109 +0xd5
2021-05-31T08:30:46.4475023Z github.com/nspcc-dev/neo-go/cli.newTestChain(0xc001236780, 0x0, 0x24b3788, 0x24b3788, 0x878)
2021-05-31T08:30:46.4476093Z 	/home/runner/work/neo-go/neo-go/cli/executor_test.go:80 +0x75f
2021-05-31T08:30:46.4477216Z github.com/nspcc-dev/neo-go/cli.newExecutorWithConfig(0xc001236780, 0x1, 0x0, 0x109983b)
2021-05-31T08:30:46.4478380Z 	/home/runner/work/neo-go/neo-go/cli/executor_test.go:104 +0x525
2021-05-31T08:30:46.4479293Z github.com/nspcc-dev/neo-go/cli.newExecutor(...)
2021-05-31T08:30:46.4480225Z 	/home/runner/work/neo-go/neo-go/cli/executor_test.go:91
2021-05-31T08:30:46.4481266Z github.com/nspcc-dev/neo-go/cli.TestImportDeployed(0xc001236780)
2021-05-31T08:30:46.4482320Z 	/home/runner/work/neo-go/neo-go/cli/wallet_test.go:243 +0x65
2021-05-31T08:30:46.4482959Z testing.tRunner(0xc001236780, 0x1b90390)
2021-05-31T08:30:46.4483701Z 	/opt/hostedtoolcache/go/1.15.12/x64/src/testing/testing.go:1123 +0x203
2021-05-31T08:30:46.4484348Z created by testing.(*T).Run
2021-05-31T08:30:46.4485030Z 	/opt/hostedtoolcache/go/1.15.12/x64/src/testing/testing.go:1168 +0x5bc
```

### CODE REVIEW

File: pkg/wallet/account.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ type Account struct {
 	// Account import file.
 	wif string
 
+	// Scrypt KDF params n, r, p.
+	Scrypt keys.ScryptParams `json:"-"`
```
Comment (roman-khimov) [inline pkg/wallet/account.go:31]: I'd rather not add it here.

File: pkg/wallet/account.go [hunk 2]
```diff
@@ -139,7 +143,7 @@ func (a *Account) Decrypt(passphrase string) error {
 // Encrypt encrypts the wallet's PrivateKey with the given passphrase
 // under the NEP-2 standard.
 func (a *Account) Encrypt(passphrase string) error {
-	wif, err := keys.NEP2Encrypt(a.privateKey, passphrase)
+	wif, err := keys.NEP2Encrypt(a.privateKey, passphrase, a.Scrypt)
```
Comment (roman-khimov) [inline pkg/wallet/account.go:146]: You can pass scrypt params directly to `Encrypt`.

File: pkg/wallet/account.go [hunk 3]
```diff
@@ -125,7 +129,7 @@ func (a *Account) Decrypt(passphrase string) error {
 	if a.EncryptedWIF == "" {
 		return errors.New("no encrypted wif in the account")
 	}
-	a.privateKey, err = keys.NEP2Decrypt(a.EncryptedWIF, passphrase)
+	a.privateKey, err = keys.NEP2Decrypt(a.EncryptedWIF, passphrase, a.Scrypt)
```
Comment (roman-khimov) [inline pkg/wallet/account.go:132]: And to `Decrypt`, you either have a wallet you know when you use it or you're using the default settings.

File: pkg/crypto/keys/nep2.go [hunk 4]
```diff
@@ -40,15 +40,27 @@ func NEP2ScryptParams() ScryptParams {
 	}
 }
 
+// Equals checks whether n, p, r KDF parameters are equal.
+func (p ScryptParams) Equals(other ScryptParams) bool {
```
Comment (roman-khimov) [inline pkg/crypto/keys/nep2.go:44]: `Struct values are comparable if all their fields are comparable. Two struct values are equal if their corresponding non-blank fields are equal.`

**Comment by AnnaShaleva** [comment]
Well, some of these tests are still failing, but at least because of other reasons.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**